### ISSUE TITLE
NotFoundError:  Resource localhost/_AnonymousVar29/N10tensorflow22SummaryWriterInterfaceE does not exist.

### ISSUE THREAD

**Issue by GF-Huang** [tensorflow/tensorflow#47100]
tf 2.4.1
```
---------------------------------------------------------------------------
NotFoundError                             Traceback (most recent call last)
<ipython-input-10-92bfedd73a83> in <module>
      2 tensorboard_callback = tf.keras.callbacks.TensorBoard(log_dir=log_dir, histogram_freq=1, update_freq='batch')
      3 
----> 4 model.fit(x, y, initial_epoch=20, epochs=25, callbacks=[tensorboard_callback])

~/.local/lib/python3.8/site-packages/tensorflow/python/keras/engine/training.py in fit(self, x, y, batch_size, epochs, verbose, callbacks, validation_split, validation_data, shuffle, class_weight, sample_weight, initial_epoch, steps_per_epoch, validation_steps, validation_batch_size, validation_freq, max_queue_size, workers, use_multiprocessing)
   1098                 _r=1):
   1099               callbacks.on_train_batch_begin(step)
-> 1100               tmp_logs = self.train_function(iterator)
   1101               if data_handler.should_sync:
   1102                 context.async_wait()

~/.local/lib/python3.8/site-packages/tensorflow/python/eager/def_function.py in __call__(self, *args, **kwds)
    826     tracing_count = self.experimental_get_tracing_count()
    827     with trace.Trace(self._name) as tm:
--> 828       result = self._call(*args, **kwds)
    829       compiler = "xla" if self._experimental_compile else "nonXla"
    830       new_tracing_count = self.experimental_get_tracing_count()

~/.local/lib/python3.8/site-packages/tensorflow/python/eager/def_function.py in _call(self, *args, **kwds)
    853       # In this case we have created variables on the first call, so we run the
    854       # defunned version which is guaranteed to never create variables.
--> 855       return self._stateless_fn(*args, **kwds)  # pylint: disable=not-callable
    856     elif self._stateful_fn is not None:
    857       # Release the lock early so that multiple threads can perform the call

~/.local/lib/python3.8/site-packages/tensorflow/python/eager/function.py in __call__(self, *args, **kwargs)
   2940       (graph_function,
   2941        filtered_flat_args) = self._maybe_define_function(args, kwargs)
-> 2942     return graph_function._call_flat(
   2943         filtered_flat_args, captured_inputs=graph_function.captured_inputs)  # pylint: disable=protected-access
   2944 

~/.local/lib/python3.8/site-packages/tensorflow/python/eager/function.py in _call_flat(self, args, captured_inputs, cancellation_manager)
   1916         and executing_eagerly):
   1917       # No tape is watching; skip to running the function.
-> 1918       return self._build_call_outputs(self._inference_function.call(
   1919           ctx, args, cancellation_manager=cancellation_manager))
   1920     forward_backward = self._select_forward_and_backward_functions(

~/.local/lib/python3.8/site-packages/tensorflow/python/eager/function.py in call(self, ctx, args, cancellation_manager)
    553       with _InterpolateFunctionError(self):
    554         if cancellation_manager is None:
--> 555           outputs = execute.execute(
    556               str(self.signature.name),
    557               num_outputs=self._num_outputs,

~/.local/lib/python3.8/site-packages/tensorflow/python/eager/execute.py in quick_execute(op_name, num_outputs, inputs, attrs, ctx, name)
     57   try:
     58     ctx.ensure_initialized()
---> 59     tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,
     60                                         inputs, attrs, num_outputs)
     61   except core._NotOkStatusException as e:

NotFoundError:  Resource localhost/_AnonymousVar29/N10tensorflow22SummaryWriterInterfaceE does not exist.
	 [[{{node cond/then/_0/batch_loss}}]] [Op:__inference_train_function_9310]

Function call stack:
train_function
```

**Comment by ravikyram** [comment]
@GF-Huang 

Please, fill [issue template](https://github.com/tensorflow/tensorflow/issues/new/choose).
Please, share colab link or simple standalone code with supporting files to reproduce the issue in our environment.It helps us in localizing the issue faster.Thanks!

**Comment by NandeeshHD** [comment]
> tf 2.4.1
> 
> ```
> ---------------------------------------------------------------------------
> NotFoundError                             Traceback (most recent call last)
> <ipython-input-10-92bfedd73a83> in <module>
>       2 tensorboard_callback = tf.keras.callbacks.TensorBoard(log_dir=log_dir, histogram_freq=1, update_freq='batch')
>       3 
> ----> 4 model.fit(x, y, initial_epoch=20, epochs=25, callbacks=[tensorboard_callback])
> 
> ~/.local/lib/python3.8/site-packages/tensorflow/python/keras/engine/training.py in fit(self, x, y, batch_size, epochs, verbose, callbacks, validation_split, validation_data, shuffle, class_weight, sample_weight, initial_epoch, steps_per_epoch, validation_steps, validation_batch_size, validation_freq, max_queue_size, workers, use_multiprocessing)
>    1098                 _r=1):
>    1099               callbacks.on_train_batch_begin(step)
> -> 1100               tmp_logs = self.train_function(iterator)
>    1101               if data_handler.should_sync:
>    1102                 context.async_wait()
> 
> ~/.local/lib/python3.8/site-packages/tensorflow/python/eager/def_function.py in __call__(self, *args, **kwds)
>     826     tracing_count = self.experimental_get_tracing_count()
>     827     with trace.Trace(self._name) as tm:
> --> 828       result = self._call(*args, **kwds)
>     829       compiler = "xla" if self._experimental_compile else "nonXla"
>     830       new_tracing_count = self.experimental_get_tracing_count()
> 
> ~/.local/lib/python3.8/site-packages/tensorflow/python/eager/def_function.py in _call(self, *args, **kwds)
>     853       # In this case we have created variables on the first call, so we run the
>     854       # defunned version which is guaranteed to never create variables.
> --> 855       return self._stateless_fn(*args, **kwds)  # pylint: disable=not-callable
>     856     elif self._stateful_fn is not None:
>     857       # Release the lock early so that multiple threads can perform the call
> 
> ~/.local/lib/python3.8/site-packages/tensorflow/python/eager/function.py in __call__(self, *args, **kwargs)
>    2940       (graph_function,
>    2941        filtered_flat_args) = self._maybe_define_function(args, kwargs)
> -> 2942     return graph_function._call_flat(
>    2943         filtered_flat_args, captured_inputs=graph_function.captured_inputs)  # pylint: disable=protected-access
>    2944 
> 
> ~/.local/lib/python3.8/site-packages/tensorflow/python/eager/function.py in _call_flat(self, args, captured_inputs, cancellation_manager)
>    1916         and executing_eagerly):
>    1917       # No tape is watching; skip to running the function.
> -> 1918       return self._build_call_outputs(self._inference_function.call(
>    1919           ctx, args, cancellation_manager=cancellation_manager))
>    1920     forward_backward = self._select_forward_and_backward_functions(
> 
> ~/.local/lib/python3.8/site-packages/tensorflow/python/eager/function.py in call(self, ctx, args, cancellation_manager)
>     553       with _InterpolateFunctionError(self):
>     554         if cancellation_manager is None:
> --> 555           outputs = execute.execute(
>     556               str(self.signature.name),
>     557               num_outputs=self._num_outputs,
> 
> ~/.local/lib/python3.8/site-packages/tensorflow/python/eager/execute.py in quick_execute(op_name, num_outputs, inputs, attrs, ctx, name)
>      57   try:
>      58     ctx.ensure_initialized()
> ---> 59     tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,
>      60                                         inputs, attrs, num_outputs)
>      61   except core._NotOkStatusException as e:
> 
> NotFoundError:  Resource localhost/_AnonymousVar29/N10tensorflow22SummaryWriterInterfaceE does not exist.
> 	 [[{{node cond/then/_0/batch_loss}}]] [Op:__inference_train_function_9310]
> 
> Function call stack:
> train_function
> ```

@GF-Huang, have you found a solution to this issue. If so, kindly share your solution here. I'm facing similar issue.

**Comment by GF-Huang** [comment]
I forget what's the problem, maybe restart the python kernel will be help.

**Comment by BenoitLeguay** [comment]
I've got a similar issue:
_tensorflow 2.8.0_

**context**: 
For some reasons, I need to call successively 2 times `model.fit(..., callbacks=callbacks)`. 
In callbacks, I have a `tf.keras.callbacks.TensorBoard`. 
In the first `fit` everything is going well, but the second time I have this error:
```
 tf.summary.histogram('similarities', pos_scores, step=0)
    File "/Users/bleguay/.pyenv/versions/deepranker/lib/python3.10/site-packages/tensorboard/plugins/histogram/summary_v2.py", line 194, in histogram
      return tf.summary.write(
Node: 'similarities/write_summary'
Resource localhost/_AnonymousVar0/N10tensorflow22SummaryWriterInterfaceE does not exist.
         [[{{node similarities/write_summary}}]] [Op:__inference_train_function_49222]
```
Since I defined it multiple times in fit, It seems `tf.summary` has difficulty finding the current `tf.summary.SummaryWriter`. I tried to re-instantiate the `tf.keras.callbacks.TensorBoard` object before feeding it to the second fit or to remove the `callbacks` from the second `fit` but the same error occured.

How can I solve that ? what is the good practice in order to call fit multiples times and to keep the `tf.keras.callbacks.TensorBoard` working. 


If you need more information please tell me :) 
Thank you.

**Comment by BenoitLeguay** [comment]
Ok I just found [this](https://github.com/tensorflow/tensorflow/issues/24632), it seems the issue has been resolved in tensorflow 2.9.1. 
I'll update my version and keep you in touch.

**Comment by BenoitLeguay** [comment]
I managed to solve my issue using [this issue](https://github.com/tensorflow/tensorflow/issues/24632). However, it does not feel very natural to need to re-compile to model in order to solve a Tensorboard issue, should it be solved ? 
Of course, I might not understand the underlying reason, if yes I'm eager to learn it. 
Thanks !
